apps/balancer-tools/src/contexts/FormContext.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import { createContext, PropsWithChildren, useContext } from "react";
+
+import { AnalysisData, usePoolSimulator } from "./PoolSimulatorContext";
+
+interface PoolSimulatorContextType {
+  data: AnalysisData;
+  setData: (data: AnalysisData) => void;
+  isCustomData: boolean;
+}
+
+export const FormContext = createContext({} as PoolSimulatorContextType);
+
+export function InitialFormContextProvider({ children }: PropsWithChildren) {
+  const { initialData, setInitialData } = usePoolSimulator();
+
+  return (
+    <FormContext.Provider
+      value={{
+        data: initialData,
+        setData: setInitialData,
+        isCustomData: false,
+      }}
+    >
+      {children}
+    </FormContext.Provider>
+  );
+}
+
+export function CustomFormContextProvider({ children }: PropsWithChildren) {
```

Comment:
can you say more as to what made you decide to use two contexts here instead of one?